src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,105 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import React from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { PersonLink } from './PersonLink';
+import { Person } from '../types';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams<{ slug: string }>();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const searchParams = new URLSearchParams(location.search);
+  const selectedCentury = searchParams.get('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query');
+  const sex = searchParams.get('sex');
+
+  const filterPeople = (persons: Person[]) => {
+    return persons.filter(person => {
+      if (
+        selectedCentury &&
+        Math.floor(person.born / 100) + 1 !== parseInt(selectedCentury)
+      ) {
+        return false;
+      }
+
+      if (query && !person.name.toLowerCase().includes(query.toLowerCase())) {
+        return false;
+      }
+
+      if (sex && person.sex !== sex) {
+        return false;
+      }
+
+      return true;
+    });
+  };
+
+  const sortPeople = (
+    filteredPeople: Person[],
+    sortField: string,
+    sortOrder: string,
+  ) => {
+    return people.sort((a, b) => {
+      const aValue = a[sortField as keyof Person];
+      const bValue = b[sortField as keyof Person];
+
+      if (aValue < bValue) {
+        return sortOrder === 'asc' ? -1 : 1;
+      }
+
+      if (aValue > bValue) {
+        return sortOrder === 'asc' ? 1 : -1;
+      }
+
+      return 0;
+    });
```

Comment:
The `sortPeople` function is sorting the `people` array directly, which can lead to unexpected side effects since it mutates the original array. Consider creating a copy of the array before sorting to avoid mutating the original data.